src/map/mob.c

```diff
@@ -5965,6 +6000,11 @@ static void mob_reload(void)
 			item_drop_ratio_db[i] = NULL;
 		}
 	}
+	for (i = 0; i < MOBG_MAX_GROUP; i++) {
+		VECTOR_CLEAR(*mob->mob_groups[i]);
+		aFree(mob->mob_groups[i]);
```

Comment:
This might be attempting to free a NULL pointer if a specific group was never initialized.

But does this have to be an array of pointers? Have you considered making it an array of structs (i.e. `struct mob_group mob_groups[MOBG_MAX_GROUP];`)? Any drawbacks?

Reply:
Initially I wanted to remove the limit which is why it's a pointer, but with the current use cases you're right an array of structs is more appropriate.